### DISCUSSION TITLE
Replacement for FrontendAuthorizationClient

### DISCUSSION THREAD

**Discussion by hanan-commits-work** [iTwin/itwinjs-core#2984] [Q&A]
In the 3.0 update, frontendAuthorizationClient has been completely removed, what is the work around for it?
Should we use BrowserAuthorizationClient? If not, then what else?

**Comment by ramanujam-raman-zz** [comment]
If this is a case for browser based authorization, you need to use BrowserAuthorizationClient instead of OidcBrowserClient. 

Note that OidcBrowserClient was deprecated in 2.0 and replaced with BrowserAuthorizationClient - the change is documented here - https://www.itwinjs.org/changehistory/2.0.0/#authorization-in-browsers

In 3.0, the utilty has been moved to it's own package (@itwin/browser-authorization) - 
https://github.com/iTwin/itwinjs-core/blob/master/docs/changehistory/NextVersion.md

The field IModelApp.authorizationClient is now just set to AuthorizationClient instead of FrontendAuthorizationClient - BrowserAuthorizationClient implements AuthorizationClient.

**Reply by hanan-commits-work** [reply]
I am trying to implement the token server in 3.0, similar to this repo https://github.com/calebmshafer/itwin-viewer-proxy
so i will be using agent based authorization and not browser. What should be the approach for this?

**Reply by hanan-commits-work** [reply]
@calebmshafer can you help me with making this change?

**Reply by aruniverse** [reply]
You're going to want to use [@itwin/service-authorization](https://www.npmjs.com/package/@itwin/service-authorization). Will look into updating that viewer-proxy sample

**Reply by hanan-commits-work** [reply]
I will look into it. And what would be the replacement for oidcClient?
I passed authConfig={{ oidcClient: NoSignInIAuthClient.oidcClient  }} as an argument for <Viewer>. 
But the react-viewer is also being updated and I am using 2.0.0-dev. I tried following a similar approach, but the new viewer only seem to take BrowserClientConfiguration object as the argument for authConfig. 
What should be the work around for this?

**Reply by aruniverse** [reply]
That's incorrect, it doesn't only accept a BrowserAuthConfig. In 3.0 the viewer expects an auth client that implements our auth interface. Take a look at https://github.com/iTwin/viewer/blob/next/NextVersion.md. There should be minimal changes to the auth client in the existing example NoSignInAuthClient
